src/app/createCompany/tests/createCompany.test.tsx

```diff
@@ -0,0 +1,306 @@
+import { vi } from 'vitest';
+
+import { createImageMock } from '@/app/testsUtils/createImageMock';
+import { createCompany } from '@/app/utils/api/createCompany';
+import { render, screen, cleanup } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import CreateCompany from '../page';
+
+vi.mock(`next/navigation`, () => {
+  const actual = vi.importActual(`next/navigation`);
+  return {
+    ...actual,
+    useRouter: vi.fn(() => ({
+      push: vi.fn(),
+    })),
+  };
+});
+
+vi.mock(`@clerk/nextjs`, () => ({
+  __esModule: true,
+  useUser: () => ({
+    user: {
+      id: `1`,
+      emailAddresses: [
+        {
+          emailAddress: `test@example.com`,
+        },
+      ],
+    },
+  }),
+}));
+
+vi.mock(`@/app/utils/api/createCompany`);
+vi.mocked(createCompany).mockImplementation(async () => ({
+  companyId: `mockId`,
+}));
+
+describe(`CreateCompany`, () => {
+  afterAll(() => {
+    vi.clearAllMocks();
+  });
+
+  afterEach(() => {
+    cleanup();
+    vi.resetAllMocks();
+  });
+
+  it(`When trying to send form without filling the company name input should render error`, async () => {
+    render(<CreateCompany />);
+
+    const submitButton = screen.getByRole(`button`, {
+      name: `Finalizar cadastro`,
+    });
+
+    await userEvent.click(submitButton);
+
+    expect(
+      await screen.findByText(`*O Nome do restaurante é obrigatório.`)
```

Comment:
```suggestion
      await screen.findByText(`O Nome do restaurante é obrigatório.`)
```
you have to remove the "*" char of all findByTexts of the test file